### PR TITLE
add RedisCache create integration test case

### PR DESCRIPTION
**Pull request by songkaiape**

### CODE REVIEW

**Comment by vscjavaci** [comment]
Can one of the admins verify this patch?

**Comment by adashen** [comment]
retest this please

File: Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+        redisModule.refreshItems();
+
+        assertEquals(redisModule.getChildNodes().size(), 1);
+        assertEquals(redisModule.getChildNodes().get(0).getName(), config.dnsNameValue);
+        Thread.sleep(500);
+    }
+
+    public void CreateRedisCache(AzureManager azureManager, SubscriptionDetail currentSub, RedisCreateConfig config)
+            throws Exception {
+        if (!RedisCacheUtil.doValidate(azureManager, currentSub, config.dnsNameValue, config.selectedLocationValue,
```
Comment (adashen) [inline Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java]: do you want to test "doValidate" method? why we need to call this method here?

File: Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+                config.selectedResGrpValue, config.selectedPriceTierValue)) {
+            return;
+        }
+        Azure azure = azureManager.getAzure(currentSub.getSubscriptionId());
+        ProcessingStrategy processor = RedisCacheUtil.doGetProcessor(azure, skus, config.dnsNameValue,
+                config.selectedLocationValue, config.selectedResGrpValue, config.selectedPriceTierValue,
+                config.noSSLPort, config.newResGrp);
+
+        ExecutorService executor = Executors.newCachedThreadPool();
```
Comment (adashen) [inline Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java]: why need to use another thread, can we just execute in the current thread?

File: Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+        private ProcessingStrategy processor;
+
+        public CreateRedisCacheCallable(ProcessingStrategy processor) {
+            this.processor = processor;
+        }
+
+        public Void call() throws Exception {
+
+            // consume
+            processor.process();
```
Comment (adashen) [inline Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java]: just test processor.process(), there is no need to wrap a callable

File: Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+    RedisCreateConfig StdNewResGrpConfig = new RedisCreateConfig("MonaC3StdNew", "East US", "MonaC3StdNewRg",
+            priceTier.get("STD3"), false, true);
+    RedisCreateConfig StdNewResGrpConfigNonSsl = new RedisCreateConfig("MonaC4StdNewNonSsl", "East US",
+            "MonaC4StdNewNonSslRg", priceTier.get("STD4"), true, true);
+    RedisCreateConfig StdExistedResGrpConfig = new RedisCreateConfig("MonaC5StdExist", "East US", "MonaExist",
+            priceTier.get("STD5"), false, false);
+    RedisCreateConfig StdExistedResGrpConfigNonSsl = new RedisCreateConfig("MonaC6StdExistNonSsl", "East US",
+            "MonaExist", priceTier.get("STD1"), true, false);
+
+    RedisCreateConfig PremNewResGrpConfig = new RedisCreateConfig("MonaP1New", "East US", "MonaP1NewRg",
```
Comment (adashen) [inline Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java]: for each creation strategy, we need to verify whether the final created rediscache really picked such configuration

File: Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+    }
+
+    public void createRedisTest(RedisCacheModule redisModule, RedisCreateConfig config) throws Exception {
+        redisModule.refreshItems();
+        assertEquals(redisModule.getChildNodes().size(), 0);
+        CreateRedisCache(azureManagerMock, currentsub, config);
+
+        redisModule.removeAllChildNodes();
+        redisModule.refreshItems();
+
```
Comment (adashen) [inline Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java]: need to verify more stuff. The price tier, the configurations

File: Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+    RedisCreateConfig BasicExistedResGrpConfigNonSsl = new RedisCreateConfig("MonaC4BasicExistNonSsl", "East US",
+            "MonaExist", priceTier.get("BASIC4"), true, false);
+
+    RedisCreateConfig StdNewResGrpConfig = new RedisCreateConfig("MonaC3StdNew", "East US", "MonaC3StdNewRg",
+            priceTier.get("STD3"), false, true);
+    RedisCreateConfig StdNewResGrpConfigNonSsl = new RedisCreateConfig("MonaC4StdNewNonSsl", "East US",
+            "MonaC4StdNewNonSslRg", priceTier.get("STD4"), true, true);
+    RedisCreateConfig StdExistedResGrpConfig = new RedisCreateConfig("MonaC5StdExist", "East US", "MonaExist",
+            priceTier.get("STD5"), false, false);
+    RedisCreateConfig StdExistedResGrpConfigNonSsl = new RedisCreateConfig("MonaC6StdExistNonSsl", "East US",
```
Comment (jdneo) [inline Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java]: the name "MonaC6StdExistNonSsl" is not match with the price tier "STD1"

File: Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+
+    @Test
+    public void testRedisCacheCreatePremWithExistedResGrpNonSsl() throws Exception {
+        createRedisTest(redisModule, PremExistedResGrpConfigNonSsl);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        resetTest(name.getMethodName());
+
```
Comment (jdneo) [inline Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java]: This empty line can be removed

File: Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+            this.newResGrp = newResGrp;
+        }
+
+        public String dnsNameValue;
+        public String selectedResGrpValue;
+        public String selectedLocationValue;
+        public String selectedPriceTierValue;
+        public boolean noSSLPort;
+        public boolean newResGrp;
+
```
Comment (jdneo) [inline Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java]: same as above, remove this empty line.

**Pull request closed without merging**